File: Ejercicio 4/transporte.py
from abc import ABC, abstractmethod

class Vehiculo(ABC):
    def __init__(self, nombre):
        self.nombre = nombre

    @abstractmethod
    def calcular_tarifa_minima(self):
        pass

class Bus(Vehiculo):
    def calcular_tarifa_minima(self):
        return 4000

class Taxi(Vehiculo):
    def calcular_tarifa_minima(self):
        return 6000

class Uber(Vehiculo):
    def calcular_tarifa_minima(self):
        return 10000



vehiculos = [
    Bus("Bus"),
    Taxi("Taxi"),
    Uber("Uber")
]

for v in vehiculos:
    print(f"{v.nombre}: tarifa minima = {v.calcular_tarifa_minima()}")

File: Ejercicio 4/transporte.java
abstract class Vehiculo {
    protected String nombre;

    public Vehiculo(String nombre) {
        this.nombre = nombre;
    }

    public abstract double calcularTarifaMinima();
}

class Bus extends Vehiculo {
    public Bus(String n) {
        super(n);
    }

    public double calcularTarifaMinima() {
        return 4000;
    }
}

class Taxi extends Vehiculo {
    public Taxi(String n) {
        super(n);
    }

    public double calcularTarifaMinima() {
        return 6000;
    }
}

class Uber extends Vehiculo {
    public Uber(String n) {
        super(n);
    }

    public double calcularTarifaMinima() {
        return 10000;
    }
}

public class Main {
    public static void main(String[] args) {
        Vehiculo[] vehiculos = {
            new Bus("Bus"),
            new Taxi("Taxi"),
            new Uber("Uber")
        };

        for (Vehiculo v : vehiculos) {
            System.out.println(v.nombre + ": tarifa minima= " + v.calcularTarifaMinima());
        }
    }
}
